src/main/java/Core/Minimax.java

```diff
@@ -7,70 +7,77 @@
 public class Minimax {
 
     private Grid firstGrid;
-    private Mark minimaxPlayerMark;
-    private Mark nonMinimaxPlayerMark;
+    private Mark maximisingPlayerMark;
+    private Mark minimisingPlayerMark;
 
-    public Minimax(Grid firstGrid, Mark minimaxPlayerMark, Mark nonMinimaxPlayerMark) {
+    public Minimax(Grid firstGrid, Mark maximisingPlayerMark, Mark minimisingPlayerMark) {
         this.firstGrid = firstGrid;
-        this.minimaxPlayerMark = minimaxPlayerMark;
-        this.nonMinimaxPlayerMark = nonMinimaxPlayerMark;
+        this.maximisingPlayerMark = maximisingPlayerMark;
+        this.minimisingPlayerMark = minimisingPlayerMark;
     }
 
-    public int findSquareChoice(Grid grid, Mark optimisingPlayerMark, Integer depth) {
-
-        ArrayList<Integer> emptyGridSquares = grid.emptySquareIndices();
+    public int findGridChoice() {
+        ArrayList<Integer> emptyGridSquares = firstGrid.emptySquareIndices();
         ArrayList<Integer> scores = new ArrayList<>();
 
-        emptyGridSquares
-                .stream()
-                .map(emptySquare -> emulateGrid(grid, emptySquare, optimisingPlayerMark))
-                .forEach(emulatedGrid ->
-                {
-                    Integer emulatedGridScore = findGridScore(emulatedGrid, optimisingPlayerMark, depth);
-                    scores.add(emulatedGridScore);
-                });
+        for (Integer emptySquare : emptyGridSquares) {
+            Grid emulatedGrid = emulateGrid(firstGrid, emptySquare, maximisingPlayerMark);
+            Integer score = minimax(emulatedGrid, minimisingPlayerMark, 0, Integer.MIN_VALUE, Integer.MAX_VALUE);
+            scores.add(score);
+        }
 
-        Integer relevantIndex;
-        if (grid != firstGrid) {
-            if (optimisingPlayerMark == minimaxPlayerMark) {
-                relevantIndex = findMaxIndex(scores);
-            } else {
-                relevantIndex = findMinIndex(scores);
-            }
-            return scores.get(relevantIndex);
+        Integer maxScoreIndex = findMaxIndex(scores);
+        return emptyGridSquares.get(maxScoreIndex);
```

Comment:
You're finding the possible moves, deciding which one is best, then finding the index of that score, then mapping that back to the possible moves. Feels like an implementation detail is leaking into other parts of the code. Can you think of a way to avoid this?